src/test/incremental/issue-54242.rs

```diff
@@ -1,6 +1,6 @@
 // revisions: rpass cfail
 
-trait Tr {
+trait Tr where Self::Arr: Sized {
```

Comment:
Uh..?

Reply:
So, the cycle is broken with this PR because bounds don't need to be proven to prove that the trait is well-formed. This makes the code work like it does currently.